src/components/slider/slider.component.ts

```diff
@@ -0,0 +1,776 @@
+
+        // check if we should show ticks
+        if (majorOptions.show === false && minorOptions.show === false) {
+            this.ticks = [];
+        }
+
+        // create ticks for both major and minor
+        let majorTicks = this.getTicks(majorOptions, SliderTickType.Major);
+        let minorTicks = this.getTicks(minorOptions, SliderTickType.Minor);
+
+        // remove any minor ticks that are on a major interval
+        this.ticks = this.unionTicks(majorTicks, minorTicks);
+    }
+
+    private updateTrackColors() {
+
+        // get colors for each part of the track
+        let lower = this.options.track.colors.lower;
+        let range = this.options.track.colors.range;
+        let higher = this.options.track.colors.higher;
+
+        // update the controller value
+        this.tracks.lower.color = typeof lower === 'string' ? lower : `linear-gradient(to right, ${lower.join(', ')})`;
+        this.tracks.middle.color = typeof range === 'string' ? range : `linear-gradient(to right, ${range.join(', ')})`;
+        this.tracks.upper.color = typeof higher === 'string' ? higher : `linear-gradient(to right, ${higher.join(', ')})`;
+    }
+
+    private getSteps(steps: number | number[]): number[] {
+
+        // if they are already an array just return it
+        if (steps instanceof Array) {
+            return steps;
+        }
+
+        let output: number[] = [];
+
+        // otherwise calculate the steps
+        for (let idx = this.options.track.min; idx <= this.options.track.max; idx += steps) {
+            output.push(idx);
+        }
+
+        return output;
+    }
+
+    private getTicks(options: SliderTickOptions, type: SliderTickType): SliderTick[] {
+
+        // create an array to store the ticks and step points
+        let steps = this.getSteps(options.steps);
+
+        // get some chart options
+        let min = this.options.track.min;
+        let max = this.options.track.max;
+
+        // convert each step to a slider tick and remove invalid ticks
+        return steps.map(step => {
+            return {
+                showTicks: options.show,
+                showLabels: options.labels,
+                type: type,
+                position: ((step - min) / (max - min)) * 100,
+                value: step,
+                label: options.formatter(step)
+            };
+        }).filter(tick => tick.position >= 0 && tick.position <= 100);
+    }
+
+    private unionTicks(majorTicks: SliderTick[], minorTicks: SliderTick[]): SliderTick[] {
+
+        // get all ticks combined removing any minor ticks with the same value as major ticks
+        return majorTicks.concat(minorTicks)
+            .filter((tick, index, array) => tick.type === SliderTickType.Major || !array.find(tk => tk.type === SliderTickType.Major && tk.position === tick.position))
+            .sort((t1, t2) => t1.value - t2.value);
+    }
+
+    private deepMerge(destination: any, source: any) {
```

Comment:
Would these be worth adding to Object or otherwise factoring out to be globally available?